qtpy/tests/test_missing_optional_deps.py

```diff
@@ -0,0 +1,20 @@
+"""Test the deferred import error mechanism"""
+
+
+# See https://github.com/spyder-ide/qtpy/pull/387/
+
+
+import pytest
+
+
+def test_missing_optional_deps():
+    """Test importing a module that uses the deferred import error mechanism"""
+    from . import optional_deps
+
+    assert optional_deps.ExampleClass is not None
+
+    with pytest.raises(ImportError) as excinfo:
```

Comment:
```suggestion
    with pytest.raises(QtModuleNotInstalledError) as excinfo:
```

We should check the actual error class here, no?